internal/webhook/forwarder.go

```diff
@@ -0,0 +1,139 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"strings"
+
+	"hubproxy/internal/storage"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	webhookForwardedEvents = promauto.NewCounter(
+		prometheus.CounterOpts{
+			Name: "hubproxy_webhook_forwarded_events_total",
+			Help: "Total number of webhook events forwarded to the target",
+		},
+	)
+
+	webhookForwardingErrors = promauto.NewCounter(
+		prometheus.CounterOpts{
+			Name: "hubproxy_webhook_forwarding_errors_total",
+			Help: "Total number of webhook forwarding errors",
+		},
+	)
+)
+
+type WebhookForwarder struct {
+	targetURL  string
+	httpClient *http.Client
+	storage    storage.Storage
+	logger     *slog.Logger
+	queue      chan struct{}
+}
+
+func (f *WebhookForwarder) forwardEvent(ctx context.Context, event *storage.Event) {
+	var targetURL string
+	// http.NewRequest still needs a valid http URI, make a fake one for unix socket path
+	if strings.HasPrefix(f.targetURL, "unix://") {
+		targetURL = "http://127.0.0.1/webhook"
+	} else {
+		targetURL = f.targetURL
+	}
+
+	req, err := http.NewRequest(http.MethodPost, targetURL, strings.NewReader(string(event.Payload)))
+	if err != nil {
+		webhookForwardingErrors.Inc()
+		f.logger.Error("failed to create request", "targetURL", targetURL, "error", err)
+		return
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-GitHub-Event", event.Type)
+	req.Header.Set("X-GitHub-Delivery", event.ID)
+	// TODO: Where is X-Hub-Signature-256 stored in the db?
+	// req.Header.Set("X-Hub-Signature-256", event.Signature)
```

Comment:
@tnm Do you know if we store the original signature somewhere? Before we could get it from the original request right from memory, but if we're processing it later, we'll need to pull it from the db somehow.

Reply:
#42

Reply:
#58